src/nvim/option.c

```diff
@@ -3642,69 +3639,76 @@ char *get_varp_scope(vimoption_T *p, int scope)
     case PV_MP:
-      return (char *)&(curbuf->b_p_mp);
+      return (char *)&(buf->b_p_mp);
     case PV_EP:
-      return (char *)&(curbuf->b_p_ep);
+      return (char *)&(buf->b_p_ep);
     case PV_KP:
-      return (char *)&(curbuf->b_p_kp);
+      return (char *)&(buf->b_p_kp);
     case PV_PATH:
-      return (char *)&(curbuf->b_p_path);
+      return (char *)&(buf->b_p_path);
     case PV_AR:
-      return (char *)&(curbuf->b_p_ar);
+      return (char *)&(buf->b_p_ar);
     case PV_TAGS:
-      return (char *)&(curbuf->b_p_tags);
+      return (char *)&(buf->b_p_tags);
     case PV_TC:
-      return (char *)&(curbuf->b_p_tc);
+      return (char *)&(buf->b_p_tc);
     case PV_SISO:
-      return (char *)&(curwin->w_p_siso);
+      return (char *)&(win->w_p_siso);
     case PV_SO:
-      return (char *)&(curwin->w_p_so);
+      return (char *)&(win->w_p_so);
     case PV_DEF:
-      return (char *)&(curbuf->b_p_def);
+      return (char *)&(buf->b_p_def);
     case PV_INC:
-      return (char *)&(curbuf->b_p_inc);
+      return (char *)&(buf->b_p_inc);
     case PV_DICT:
-      return (char *)&(curbuf->b_p_dict);
+      return (char *)&(buf->b_p_dict);
     case PV_TSR:
-      return (char *)&(curbuf->b_p_tsr);
+      return (char *)&(buf->b_p_tsr);
     case PV_TSRFU:
-      return (char *)&(curbuf->b_p_tsrfu);
+      return (char *)&(buf->b_p_tsrfu);
     case PV_TFU:
-      return (char *)&(curbuf->b_p_tfu);
+      return (char *)&(buf->b_p_tfu);
     case PV_SBR:
-      return (char *)&(curwin->w_p_sbr);
+      return (char *)&(win->w_p_sbr);
     case PV_STL:
-      return (char *)&(curwin->w_p_stl);
+      return (char *)&(win->w_p_stl);
     case PV_WBR:
-      return (char *)&(curwin->w_p_wbr);
+      return (char *)&(win->w_p_wbr);
     case PV_UL:
-      return (char *)&(curbuf->b_p_ul);
+      return (char *)&(buf->b_p_ul);
     case PV_LW:
-      return (char *)&(curbuf->b_p_lw);
+      return (char *)&(buf->b_p_lw);
     case PV_BKC:
-      return (char *)&(curbuf->b_p_bkc);
+      return (char *)&(buf->b_p_bkc);
     case PV_MENC:
-      return (char *)&(curbuf->b_p_menc);
+      return (char *)&(buf->b_p_menc);
     case PV_FCS:
-      return (char *)&(curwin->w_p_fcs);
+      return (char *)&(win->w_p_fcs);
     case PV_LCS:
-      return (char *)&(curwin->w_p_lcs);
+      return (char *)&(win->w_p_lcs);
     case PV_VE:
-      return (char *)&(curwin->w_p_ve);
+      return (char *)&(win->w_p_ve);
     }
     return NULL;     // "cannot happen"
   }
   return (char *)get_varp(p);
 }
 
-/// Get pointer to option variable.
-static char_u *get_varp(vimoption_T *p)
+/// Get pointer to option variable, depending on local or global scope.
+///
+/// @param scope  can be OPT_LOCAL, OPT_GLOBAL or a combination.
+char *get_varp_scope(vimoption_T *p, int scope)
+{
+  return get_varp_scope_from(p, scope, curbuf, curwin);
+}
+
+static char_u *get_varp_from(vimoption_T *p, buf_T *buf, win_T *win)
```

Comment:
shouldn't we get `buf` from `win` ?

Reply:
This is driven from `nvim_[gs]set_option_value` which allows specifying a specific buffer/window.